packages/shared/src/utils/validation/declarative-validation-rules.ts

```diff
@@ -1,37 +1,46 @@
 import { addComma } from '../currency';
 import { cloneObject } from '../object';
 import { compareBigUnsignedInt } from '../string';
+import { TFormErrorMessagesTypes } from './form-error-messages-types';
 
-const validRequired = (value /* , options, field */) => {
+export type TOptions = {
+    min: number;
+    max: number;
+    type?: string;
+    decimals?: string | number;
+    regex?: RegExp;
+};
+
+const validRequired = (value?: string /* , options, field */) => {
     if (value === undefined || value === null) {
         return false;
     }
 
-    const str = String(value).replace(/\s/g, '');
+    const str = value.replace(/\s/g, '');
     return str.length > 0;
 };
-export const validAddress = value => !/[`~!$%^&*_=+[}{\]\\"?><|]+/.test(value);
-export const validPostCode = value => value === '' || /^[A-Za-z0-9][A-Za-z0-9\s-]*$/.test(value);
-export const validTaxID = value => /(?!^$|\s+)[A-Za-z0-9./\s-]$/.test(value);
-export const validPhone = value => /^\+?([0-9-]+\s)*[0-9-]+$/.test(value);
-export const validLetterSymbol = value => /^[A-Za-z]+([a-zA-Z\.' -])*[a-zA-Z\.' -]+$/.test(value);
-export const validLength = (value = '', options) =>
+export const validAddress = (value: string) => !/[`~!$%^&*_=+[}{\]\\"?><|]+/.test(value);
+export const validPostCode = (value: string) => value === '' || /^[A-Za-z0-9][A-Za-z0-9\s-]*$/.test(value);
+export const validTaxID = (value: string) => /(?!^$|\s+)[A-Za-z0-9./\s-]$/.test(value);
+export const validPhone = (value: string) => /^\+?([0-9-]+\s)*[0-9-]+$/.test(value);
+export const validLetterSymbol = (value: string) => /^[A-Za-z]+([a-zA-Z\.' -])*[a-zA-Z\.' -]+$/.test(value);
```

Comment:
```suggestion
export const validLetterSymbol = (value: string) => /^[A-Za-z]+([a-zA-Z.' -])*[a-zA-Z.' -]+$/.test(value);
```
_Unnecessary escape character: \.._